### PR TITLE
Added pa11y testing to the site using GitHub action.

### LINKED ISSUE #103: Move on from a11y
Looks like https://github.com/addyosmani/a11y has been archived and is no longer receiving security updates. Evaluate whether to replace with something like https://github.com/pa11y/pa11y or revisit how we want to provide 508 compliance/assistance.

### PR DESCRIPTION
**Pull request by dmundra**
You can see an example of how it runs over in my fork: https://github.com/dmundra/federalist-uswds-jekyll/pull/1/checks

The GitHub action is built to run on pull requests but can be modified to run on pushes or schedules. I believe this can also be ported to CircleCI.

Another example: https://github.com/18F/accessibility/pull/270 and full process is documented here https://accessibility.civicactions.com/posts/automated-accessibility-testing-leveraging-github-actions-and-pa11y-ci-with-axe

Let me know if have questions and recommendations.

Closes #103.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,14 +6,17 @@
     "clean": "bundle exec jekyll clean",
     "reset": "npx rimraf .git",
     "start": "bundle exec jekyll serve",
-    "test": "bundle exec htmlproofer _site; npx a11y '_site/**/*.html'"
+    "start-detached": "bundle exec jekyll serve --detach",
+    "test": "bundle exec htmlproofer _site; npx a11y '_site/**/*.html'",
```
Comment (DaveSkender) [inline package.json:10]: If you're replacing `a11y`, there are some references here that you'll need to remove also.

Once you remove the offending `a11y` package and this extraneous ` npx a11y '_site/**/*.html'` from `test` here, you may uncover a hidden pre-existing problem with `htmlproofer` where it's producing false-negative test results due to differences in `site.url` vs `localhost` internal URLs.

At least, that's what happening on my downstream site.  I'm still troubleshooting, but interested to see what you do here.

Comment (dmundra) [reply package.json]: I went ahead and removed `a11y` and rebased this PR. I ran the `npm test` locally and it seemed to pass for me.

Comment (DaveSkender) [reply package.json]: I think the problems start when you set a `site.url` in `_config.yml`.

Comment (dmundra) [reply package.json]: Right and for another Jekyll site (https://github.com/CivicActions/accessibility/blob/main/package.json#L13) we replace the live site URL with the internal one for pa11y-ci.

htmlproofer does have a `url_swap` option that might be able to do the same thing.

Comment (dmundra) [reply package.json]: The one thing I realized is that package.json won't be able to get the site URL so you have to manage your own versions with the specific site URLs.

Comment (DaveSkender) [reply package.json]: For `htmlproofer`, I was able troubleshoot this a bit more.  This is caused by the `meta` absolute reference in `<link rel="canonical" href="{{ page.url | absolute_url }}" />` that is produced in the `build`.  There's two solutions:

1) use a custom swap in the `htmlproofer` command line, to basically remove `site.url` name: `--url-swap "https\://digitalcorps.gsa.gov:"`.

2) add an ignore statement in the `canonical`:

```html
 <link rel="canonical" href="{{ page.url | absolute_url }}" data-proofer-ignore />
```

**Comment by apburnes** [comment]
Thanks the contributions @dmundra! I'm going to move this PR 's destination from main branch into another feature branch so we are able to let the site build on Federalist and feel out the action. Thanks again for you contributions and thanks @DaveSkender for your comments.

👍 👍

**Comment by dmundra** [comment]
Thanks, @apburnes! That sounds like a good plan.

**Pull request merged**